build-tools/packages/build-cli/src/handlers/fluidReleaseStateHandler.ts

```diff
@@ -176,6 +180,16 @@ export class FluidReleaseStateHandler extends InitFailedStateHandler {
 				break;
 			}
 
+			case "CheckReleaseNotes": {
```

Comment:
Not in this PR, but consider implementing this switch statement as a map from MachineState to function. It would remove a lot of the boilerplate since you wouldn't need the actual argument list, case and break for each one.

Reply:
Such an approach would make things like copying the map but replacing a few problematic callbacks with mocks easy for testing.

Reply:
Doen in https://github.com/microsoft/FluidFramework/pull/22929